tests/integration/redis_tests.py

```diff
@@ -68,3 +73,49 @@ def test_pipeline(self):
         self.assertTrue(span_observer.on_start_called)
         self.assertTrue(span_observer.on_finish_called)
         self.assertIsNone(span_observer.on_finish_exc_info)
+
+
+class RedisMessageQueueTests(unittest.TestCase):
+    qname = "redisTestQueue"
+
+    def setUp(self):
+        self.pool = redis.ConnectionPool(host="localhost")
+
+    def test_put_get(self):
+        message_queue = MessageQueue(self.qname, self.pool)
+
+        with contextlib.closing(message_queue) as mq:
+            mq.put(b"x")
+            message = mq.get()
+            self.assertEqual(message, b"x")
+
+    def test_get_timeout(self):
+        message_queue = MessageQueue(self.qname, self.pool)
+
+        with contextlib.closing(message_queue) as mq:
+            start = time.time()
+            with self.assertRaises(TimedOutError):
+                message_queue.get(timeout=1)
+            elapsed = time.time() - start
+            self.assertAlmostEqual(elapsed, 1.0, places=0)
```

Comment:
Yeah, I did run into some failures for the equivalent test in the POSIX variant for precision issues on some cases.  It likely could be due to abstraction layers (I was running them inside a container) but it may be worth being aware of.